wallet-test/src/test/scala/org/bitcoins/wallet/util/BitcoinSWalletTest.scala

```diff
@@ -57,21 +58,44 @@ trait BitcoinSWalletTest
       .map(_ => ())
   }
 
-  def createNewWallet(): Future[UnlockedWalletApi] = {
-
-    for {
-      _ <- config.initialize()
-      wallet <- Wallet.initialize().map {
-        case InitializeWalletSuccess(wallet) => wallet
-        case err: InitializeWalletError =>
-          logger.error(s"Could not initialize wallet: $err")
-          fail(err)
-      }
-    } yield wallet
-  }
+  private val createNewWallet: Option[Config] => () => Future[
+    UnlockedWalletApi] =
+    extraConfig =>
+      () => {
+        val defaultConf = implicitly[WalletAppConfig]
+        val walletConfig = extraConfig match {
+          case None    => defaultConf
+          case Some(c) => defaultConf.withOverrides(c)
+        }
+
+        AppConfig.throwIfDefaultDatadir(walletConfig)
+
+        for {
+          _ <- walletConfig.initialize()
+          wallet <- Wallet
+            .initialize()(implicitly[ExecutionContext], walletConfig)
+            .map {
+              case InitializeWalletSuccess(wallet) => wallet
```

Comment:
There is where i think for expressions get really tough. I think we should be brekaing this out so it is more readable. Either create a function of use `.map` IMO